### PR TITLE
Update prog-dx-with-com.md

### PR DESCRIPTION
**Pull request by walbourn**
Added more information about smart-pointer choices, link libraries, and the IID_PPV_ARGS macro

### CODE REVIEW

**Comment by PRMerger7** [comment]
@walbourn : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: desktop-src/prog-dx-with-com.md [hunk 1]
```diff
@@ -57,6 +57,9 @@ Although a GUID is a structure, a GUID is often expressed in equivalent string f
 
 Because the actual GUID is somewhat clumsy to use and easy to mistype, an equivalent name is usually provided as well. In your code, you can use this name instead of the actual structure when you call functions, for example when you pass an argument for the `riid` parameter to [**D3D12CreateDevice**](/windows/desktop/api/d3d12/nf-d3d12-d3d12createdevice). The customary naming convention is to prepend either IID_ or CLSID_ to the descriptive name of the interface or object, respectively. For example, the name of the **ID3D12Device** interface's IID is IID_ID3D12Device.
 
+> [!NOTE]
+> DirectX applications should link with ``dxguid.lib`` and ``uuid.lib`` to provide definitions for the varoius interface and class GUIDs. Visual C++ and other compilers support the **__uuidof** operator language extension, but explicilt C-style linkage with these link libraries is also supported and fully portable.
```
Comment (cremno) [inline desktop-src/prog-dx-with-com.md]: >explicilt

typo

File: desktop-src/prog-dx-with-com.md [hunk 2]
```diff
@@ -211,6 +214,10 @@ hr = pDXGISwapChain1->QueryInterface(IID_IDXGISwapChain3, (LPVOID*)&pDXGISwapCha
 if (FAILED(hr)) return hr;
 ```
 
+> [!NOTE]
+> In C++ you can make use of the ``IID_PPV_ARGS`` macro rather thant the explicit IID and cast pointer: ``pDXGISwapChain1->QueryInterface(IID_PPV_ARGS(&pDXGISwapChain3));``.
```
Comment (cremno) [inline desktop-src/prog-dx-with-com.md]: >thant

**Review by cremno — CHANGES_REQUESTED** [review]

**Comment by PRMerger8** [comment]
@walbourn : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Review by stevewhims — APPROVED** [review]

**Pull request merged**